database/migrations/default/1726766736889_patch-units_cris_insert_rows/down.sql

```diff
@@ -0,0 +1,78 @@
+--revert to migrations/default/1725990797742_veh_hnr_fl - which is bugged out!
+CREATE OR REPLACE FUNCTION public.units_cris_insert_rows()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+BEGIN
+    -- insert new combined / official record
+    INSERT INTO public.units (
+        autonomous_unit_id,
+        contrib_factr_1_id,
+        contrib_factr_2_id,
+        contrib_factr_3_id,
+        contrib_factr_p1_id,
+        contrib_factr_p2_id,
+        crash_pk,
+        created_by,
+        cris_crash_id,
+        e_scooter_id,
+        first_harm_evt_inv_id,
+        id,
+        is_deleted,
+        pbcat_pedalcyclist_id,
+        pbcat_pedestrian_id,
+        pedalcyclist_action_id,
+        pedestrian_action_id,
+        rpt_autonomous_level_engaged_id,
+        unit_desc_id,
+        unit_nbr,
+        updated_by,
+        veh_body_styl_id,
+        veh_damage_description1_id,
+        veh_damage_description2_id,
+        veh_damage_direction_of_force1_id,
+        veh_damage_direction_of_force2_id,
+        veh_damage_severity1_id,
+        veh_damage_severity2_id,
+        veh_hnr_fl,
+        veh_make_id,
+        veh_mod_id,
+        veh_mod_year,
+        veh_trvl_dir_id,
+        vin
+    ) values (
+        new.autonomous_unit_id,
+        new.contrib_factr_1_id,
+        new.contrib_factr_2_id,
+        new.contrib_factr_3_id,
+        new.contrib_factr_p1_id,
+        new.contrib_factr_p2_id,
+        new.crash_pk,
+        new.created_by,
+        new.cris_crash_id,
+        new.e_scooter_id,
+        new.first_harm_evt_inv_id,
+        new.id,
+        new.is_deleted,
+        new.pbcat_pedalcyclist_id,
+        new.pbcat_pedestrian_id,
+        new.pedalcyclist_action_id,
+        new.pedestrian_action_id,
+        new.rpt_autonomous_level_engaged_id,
+        new.unit_desc_id,
+        new.unit_nbr,
+        new.updated_by,
+        new.veh_body_styl_id,
+        new.veh_damage_description1_id,
+        new.veh_damage_description2_id,
+        new.veh_damage_direction_of_force1_id,
+        new.veh_damage_direction_of_force2_id,
+        new.veh_damage_severity1_id,
+        new.veh_damage_severity2_id,
+        new.veh_make_id,
+        new.veh_hnr_fl,
```

Comment:
you can see the bug here in the down migration `new.veh_make_id` and `new.veh_hnr_fl` were swapped. this is a terrifying issue if, say, both of these were boolean types. it would have been very hard to catch that we were routing data to the wrong columns 😓

Reply:
we could absolutely write a helper script that generates this function based on the `_column_metadata` table. that is probably the right move 🤔

Reply:
oh yeah! this is so easy to overlook. i scanned the order of the insert and value fields in the new migration, and I see everything matching up! 🙌

Reply:
+1 to your idea of automating the creation of this (or any) function. code that writes code is always the hardest to read and write, but good gracious is it handy to have.